provider/equinix/environ_network.go

```diff
@@ -0,0 +1,199 @@
+// Copyright 2020 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package equinix
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/juju/errors"
+	"github.com/juju/juju/core/instance"
+	"github.com/juju/juju/core/network"
+	"github.com/juju/juju/environs"
+	"github.com/juju/juju/environs/context"
+	"github.com/juju/juju/environs/instances"
+	"github.com/juju/names/v4"
+)
+
+var _ environs.Networking = (*environ)(nil)
+
+// Subnets returns basic information about subnets known by the provider for
+// the environment.
+func (e *environ) Subnets(ctx context.ProviderCallContext, inst instance.Id, subnetIDs []network.Id) ([]network.SubnetInfo, error) {
+	projectID := e.cloud.Credential.Attributes()["project-id"]
+	ips, _, err := e.equinixClient.ProjectIPs.List(projectID, nil)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	var projectSubnets []network.SubnetInfo
+	for _, ipblock := range ips {
+		subnetID, cidr, err := makeSubnetIDForNetwork(ipblock.ID, ipblock.Network, ipblock.CIDR)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+
+		if ipblock.Facility == nil || ipblock.Facility.Code != e.cloud.Region {
+			continue
+		}
+
+		subnet := network.SubnetInfo{
+			ProviderId:        network.Id(subnetID),
+			ProviderNetworkId: network.Id(ipblock.ID), //TODO: figure out what the network ID should be???
+			CIDR:              cidr,
+			VLANTag:           0,
```

Comment:
Do we know this?

Reply:
I don't know what it is. I can remove it if you feel like it is not needed

Reply:
Just wondering whether this is something we can populate from the packet API response.

Reply:
I have asked internally, I can't see a field that looks similar to VLANTag as part of the go client

Reply:
This is not a value we expose. At least I can't find it. not sure if @displague knows more about it

Reply:
https://github.com/juju/juju/blob/19d6a75eea61f5021a9fa0ee8e37fe7f6e3c9d53/core/network/nic.go#L88-L101

The IP Ranges that Equinix Metal manages, which 'spaces' are consuming here, are not on VLANs.

It's interesting to think how we could have Juju manage the addresses on a space, and assign those addresses to a VLAN, but I don't think we can incorporate this yet.  (The packngo fields that would map are `VirtualNetwork.ID` -> `InterfaceInfo.ProviderVLANId`, `VirtualNetwork.VXLAN` -> `InterfaceInfo.VLANTag`. https://github.com/packethost/packngo/blob/master/virtualnetworks.go#L18-L20)